File: TEST/test_cql_type.js
class SIMPLE
{
    constructor(
        )
    {
        this.Uuid = null;
        this.Bool = null;
        this.Int8 = null;
        this.Uint8 = null;
        this.Int16 = null;
        this.Uint16 = null;
        this.Int32 = null;
        this.Uint32 = null;
        this.Int64 = null;
        this.Uint64 = null;
        this.Float32 = null;
        this.Float64 = null;
        this.String = null;
        this.Date = null;
        this.DateTime = null;
        this.Blob = null;
    }
}

// ~~

class COMPOUND
{
    constructor(
        )
    {
        this.Id = null;
        this.Location = null;
        this.Name = null;
        this.NameSet = null;
        this.PhoneList = null;
        this.EmailSet = null;
        this.CompanyMap = null;
        this.SimpleDate = null;
        this.SimpleDateMap = null;
        this.SimpleDateSet = null;
        this.SimpleDateList = null;
        this.NameSetMap = null;
        this.SimplePointerArray = null;
    }
}

// ~~

class VALUE
{
    constructor(
        )
    {
        this.Id = null;
        this.Uuid = null;
        this.Name = null;
        this.Integer = null;
        this.Text = null;
    }
}

// ~~

class DATA
{
    constructor(
        )
    {
        this.Uuid = null;
        this.ValueId = null;
        this.ValueUuid = null;
    }
}

File: TEST/test_cql_type.rs
struct SIMPLE
{
    uuid : String,
    bool : bool,
    int8 : i8,
    uint8 : u8,
    int16 : i16,
    uint16 : u16,
    int32 : i32,
    uint32 : u32,
    int64 : i64,
    uint64 : u64,
    float32 : f32,
    float64 : f64,
    string : String,
    date : String,
    date_time : String,
    blob : String
}

// ~~

struct COMPOUND
{
    id : i32,
    location : String,
    name : ( String, String ),
    name_set : List< ( String, String ) >,
    phone_list : List< String >,
    email_set : List< String >,
    company_map : Map< String, String >,
    simple_date : String,
    simple_date_map : Map< ( String, String ), String >,
    simple_date_set : List< String >,
    simple_date_list : List< String >,
    name_set_map : Map< String, List< ( String, String ) > >,
    simple_pointer_array : Vec< Weak< SIMPLE > >
}

// ~~

struct VALUE
{
    id : i64,
    uuid : String,
    name : String,
    integer : i32,
    text : String
}

// ~~

struct DATA
{
    uuid : String,
    value_id : i64,
    value_uuid : String
}
